File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.x = [0]
        for i in range(len(nums)):
            self.x.append(self.x[-1] + nums[i])
        print(self.x)

    def sumRange(self, left: int, right: int) -> int:
        return self.x[right + 1] - self.x[left]


# Your NumArray object will be instantiated and called as such
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.php
class NumArray {
    /**
     * @param Integer[] $nums
     */
    function __construct($nums) {
        $this->$x = [0];
        for($i = 0; $i < sizeof($nums); $i++)
        {
            array_push($this->$x, $this->$x[sizeof($this->$x) -1] + $nums[$i]);
        }
    }
  
    /**
     * @param Integer $left
     * @param Integer $right
     * @return Integer
     */
    function sumRange($left, $right) {
        return $this->$x[$right + 1] - $this->$x[$left]; 
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * $obj = NumArray($nums);
 * $ret_1 = $obj->sumRange($left, $right);
 */